opcua.cpp

```diff
@@ -744,15 +764,25 @@ int OPCUA::subscribe()
 	if (node_ids == NULL)
 	{
 		logger->error("Failed to allocate memory for %d subscriptions", variables.size());
-		return 0;
+		return 2;
 	}
+	memset((void *) node_ids, 0, variables.size() * sizeof(char *));
+
 	int i = 0;
 	for (auto it = variables.cbegin(); it != variables.cend(); it++)
 	{
 		try
 		{
-			Node *node = new Node(m_connectionId, variables[i].c_str());
-			m_nodes.insert(pair<string, Node *>(variables[i], node));
+			// Hack: Skip any NodeId string with leading underscores in any segment of the path.
+			// ToDo: implement user configuration to skip parsing if a certain prefix is present.
+			if (it->find("._") != std::string::npos)
```

Comment:
Yes. The Kepware Simulator NodeIds consist of segments of the hierarchy separated by dots. Ashwin was looking to screen out any subset of the path that had a leading underscore because these subsets did not represent actual data. This means that anything with "._" in the path is to be skipped. This is a temporary hack. It is supposed to be replaced by a configuration feature that allows the system manager to filter out variables and/or nodes based on any prefix.